src/app/_services/api/user-actions.service.ts

```diff
@@ -0,0 +1,66 @@
+import {HttpClient, HttpParams} from '@angular/common/http';
+import {Injectable} from '@angular/core';
+import {environment} from '@environments/environment';
+import {Observable, of} from 'rxjs';
+import {catchError} from 'rxjs/operators';
+import {Action} from '@app/_models';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserActionsService {
+  private userActionUrl = new URL(
+    '/api/user-actions',
+    environment.apiBaseUrl
+  ).toString();
+
+  constructor(private http: HttpClient) {
+  }
+
+  getAllUserActions(options?): Observable<Action[]> {
+    const recent: boolean = options?.recent;
```

Comment:
do you need the question mark here? I haven't seen this syntax before!

Reply:
Yeah I learned this pattern recently, it's basically shorthand for 
"options ? options.recent : null"
since there could just be no options given